docs/api/utility-process.md

```diff
@@ -0,0 +1,128 @@
+# UtilityProcess
+
+`UtilityProcess` creates a child process with
+Node.js and Message ports enabled. It provides the equivalent of [`child_process.fork`][] API from Node.js
+but instead uses [Services API][] from Chromium to launch the child process.
+
+`UtilityProcess` is an [EventEmitter][event-emitter].
+
+## Class: UtilityProcess
+
+> Child process with Node.js integration spawned by Chromium.
+
+Process: [Main](../glossary.md#main-process)<br />
+
+### `new UtilityProcess(modulePath[, args][, options])`
+
+* `modulePath` string - Absolute path to the script that should run as entrypoint in the child process.
```

Comment:
If we're going to require that this be a path inside the app bundle (which I think is a somewhat questionable decision, but see below), then we should document that here too.

Reply:
I went with the same reasoning we used for dropping support to `--require` in the `NODE_OPTIONS` to avoid injecting untrusted scripts into the process. Given the entrypoint can only be a Javascript file, wouldn't it be better if it came from within the app bundle ? I couldn't think of an example usage that would require executing scripts from outside the application given this does not support network fetch of the entry scripts. I exposed this as a strict option but we can always relax the requirement if there is usage, didn't want to start with a relaxed api and then make it strict if any security issues were discovered in real world.

Reply:
Hm... what is the threat model here? What attack vector are you hypothesizing that this protects against? It's a bit unclear to me.

Reply:
Will be removing this check, refs https://github.com/electron/electron/pull/34980#discussion_r983884312